src/accountsdb/fuzz.zig

```diff
@@ -129,9 +129,19 @@ pub fn run(seed: u64, args: *std.process.ArgIterator) !void {
         manager_handle.join();
     }
 
-    var tracked_accounts = std.AutoArrayHashMap(Pubkey, TrackedAccount).init(allocator);
-    defer tracked_accounts.deinit();
-    try tracked_accounts.ensureTotalCapacity(10_000);
+    const Map = std.AutoArrayHashMap(Pubkey, TrackedAccount);
+    var tracked_accounts_rw = sig.sync.RwMux(Map).init(blk: {
+        var tracked_accounts = Map.init(allocator);
+        try tracked_accounts.ensureTotalCapacity(10_000);
+        break :blk tracked_accounts;
+    });
+    var map_alive = std.atomic.Value(bool).init(true);
+    defer {
+        map_alive.store(false, .seq_cst);
+        const tracked_accounts, var tracked_accounts_lg = tracked_accounts_rw.writeWithLock();
+        defer tracked_accounts_lg.unlock();
```

Comment:
Should join the spawned `readRandomAccountThread`s below. Seems like a reader could not see `map_alive=false`, get preempted before read locking, have this defer code run & deinit the map, then reader gets resumed and tries to read-lock it.

Reply:
addressed in 972fb916f818893e5c4f7f64db5ee9dbe660a322